libcloud/common/base.py

```diff
@@ -1153,10 +1153,19 @@ def __init__(self, key, secret=None, secure=True, host=None, port=None,
         self.region = region
 
         conn_kwargs = self._ex_connection_class_kwargs()
-        conn_kwargs.update({'timeout': kwargs.pop('timeout', None),
-                            'retry_delay': kwargs.pop('retry_delay', None),
-                            'backoff': kwargs.pop('backoff', None),
-                            'proxy_url': kwargs.pop('proxy_url', None)})
+
+        # Note: We do that to make sure those additional arguments which are
+        # provided via "_ex_connection_class_kwargs" are not overriden with
+        # None
+        additional_kwargs = ['timeout', 'retry_delay', 'backoff', 'proxy_url']
```

Comment:
I really dislike kwargs abuse here so I will try to move those arguments to the actual method signature and make sure it doesn't break anything.

Sadly, I think the original author had a reason to do that (putting it in the method signature might break some drivers) so maybe that won't work out...